src/inmanta/util/__init__.py

```diff
@@ -498,7 +498,7 @@ def _custom_json_encoder(o: object) -> Union[ReturnTypes, "JSONSerializable"]:
     if isinstance(o, BaseModel):
         return o.model_dump(by_alias=True)
 
-    if dataclasses.is_dataclass(o):
+    if dataclasses.is_dataclass(o) and not isinstance(o, type):
```

Comment:
Why is the second condition required?

Reply:
See [here](https://github.com/inmanta/inmanta-lsm/pull/1491/#discussion_r1399327453)